File: hackerrank/Interview Preparation Kit/Warm-up Challenges/Sales by Match/solve.cpp
/**
 *      Author: Kessyus Fofano
 *      Email: kessyus@gmail.com
**/

#include <bits/stdc++.h>
using namespace std;

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(0);

  int pair = 0;
  int n; cin >> n;
  int ar[n];
  for(auto i=0; i<n; i++){
    cin >> ar[i];
  }

  for(auto i=0; i<n-1; i++){
    if (ar[i]!=0) {
      for(auto j=i+1; j<n; j++){
        if (ar[j]!=0){
          if (ar[i] == ar[j]) {
            pair++;
            ar[j]=0;
            ar[i]=0;
          }
        }
      }
    }
  }

  cout << pair << "\n";

  return 0;
}

File: hackerrank/Algorithms/Warmup/Apple and Orange/solve.cs
/**
 *      Author: Kessyus Fofano
 *      Email: kessyus@gmail.com
**/

using System;
using System.ComponentModel;
using System.Linq;

class Solution
{

    // Complete the countApplesAndOranges function below.
    static void countApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges)
    {
        int sumApple = 0;
        int sumOrange = 0;
        for (int i = 0; i < apples.Length; i++)
        {
            if (((apples[i] + a) >= s) && ((apples[i] + a) <= t))
            {
                sumApple++;
            }
        }
        Console.WriteLine(sumApple);
        for (int i = 0; i < oranges.Length; i++)
        {
            if (((oranges[i] + b) >= s) && ((oranges[i] + b) <= t))
            {
                sumOrange++;
            }
        }
        Console.WriteLine(sumOrange);
    }

    static void Main(string[] args)
    {
        string[] st = Console.ReadLine().Split(' ');
        int s = Convert.ToInt32(st[0]);
        int t = Convert.ToInt32(st[1]);
        string[] ab = Console.ReadLine().Split(' ');
        int a = Convert.ToInt32(ab[0]);
        int b = Convert.ToInt32(ab[1]);
        string[] mn = Console.ReadLine().Split(' ');
        int m = Convert.ToInt32(mn[0]);
        int n = Convert.ToInt32(mn[1]);
        int[] apples = Array.ConvertAll(Console.ReadLine().Split(' '), applesTemp => Convert.ToInt32(applesTemp));
        int[] oranges = Array.ConvertAll(Console.ReadLine().Split(' '), orangesTemp => Convert.ToInt32(orangesTemp));
        countApplesAndOranges(s, t, a, b, apples, oranges);
    }
}

